scripts/update-rate-card-rates/update-rate-card-rates.py

```diff
@@ -73,39 +83,46 @@
     response_json = response.json()
     response_data = response_json['data']
     for rate_card in response_data:
-        rateCardHash[rate_card['name']] = {
-            'id': rate_card['id'],
-        }
-
+        if rate_card['name'] == rate_card_name:
+            rate_card_id = rate_card['id']
+            break
     next_page = response_json['next_page']
 
     if not next_page:
         break
 
 
-rates_to_send_by_rate_card = defaultdict(list)
+rates_to_send = []
 with open(csv_file_path) as csv_file:
     csv_reader = csv.DictReader(csv_file, delimiter=',')
+    fieldnames = csv_reader.fieldnames
+
+    # validate required fieldnames
+    required_fields = ["product_name", "price"]
+    known_fields = ["product_name", "price",
+                    "effective_at", "entitled", "rate_type"]
+    pricing_group_key_fields = []
+    for field in fieldnames:
+        if field not in known_fields:
+            pricing_group_key_fields.append(field)
+
+    for req_field in required_fields:
+        if req_field not in fieldnames:
+            raise ValueError(f"{req_field} field required in csv")
 
     for row in csv_reader:
-        rate_card_name = row['rate_card_name']
+        # required fields
         product_name = row['product_name']
-        effective_at = row['effective_at']
-        pricing_group_values_str = row['pricing_group_values']
-        pricing_group_values = None
-        if pricing_group_values_str:
-            pricing_group_values = json.loads(pricing_group_values_str)
-
-        entitled = row['entitled']
-        rate_type = row['rate_type']
         price = float(row['price'])
 
-        if rate_card_name not in rateCardHash:
-            print(f"Rate card {rate_card_name} not found")
-            continue
+        # optional with defaults fields
+        effective_at = row['effective_at'] if "effective_at" in fieldnames else default_effective_at
+        entitled = row['entitled'] == 'true' if "entitled" in fieldnames else True
```

Comment:
nit: earlier you had `entitled.lower()` so it might make sense to add `.lower()` here as well

Reply:
:), next time